sdcm/utils/thread.py

```diff
@@ -0,0 +1,34 @@
+import sys
+from functools import wraps
+import os
+import signal
+
+
+def background_failure_handler(signum, frame):  # pylint: disable=unused-argument
+    raise Exception("Failure/Error in background threads")
+
+
+signal.signal(signal.SIGUSR2, background_failure_handler)
+
+
+def test_failing_function(func):
+    """
+    Decorate a function that is running inside a thread,
+    when exception is raised in this function,
+    will cause the test to fail by sending a SIGUSR2 signal
+    """
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        result = None
+        _test_pid = os.getpid()
+        from sdcm.cluster import Setup
+        try:
+            result = func(*args, **kwargs)
+        except AssertionError:
+            Setup.tester_obj().result.addFailure(Setup.tester_obj(), sys.exc_info())
+            os.kill(_test_pid, signal.SIGUSR2)
+        except Exception:  # pylint: disable=broad-except
+            Setup.tester_obj().result.addError(Setup.tester_obj(), sys.exc_info())
```

Comment:
I touched some job failed, the backtrace can only be found in Jenkins console, it doesn't exists in our log files (both output.log & sct.log). Will it solve this issue?

Reply:
partly, it would should exist now as part of the test failure description.
but all of the output should be in output.log or sct.log, regardless of this change